dependencydiff/mapping.go

```diff
@@ -22,7 +22,7 @@ import (
 type ecosystem string
 
 // OSV ecosystem naming data source: https://ossf.github.io/osv-schema/#affectedpackage-field
-//nolint
+// nolint
```

Comment:
can the PR focus on the change at hand? It makes it easier to track changes in the future, as these lines PR will get added as another layer in `git blame`

renaming `asPointer` -> `asStringPointer`?

Reply:
sure!
about the renaming I was striving for consistency in the naming across the files, but I agree with the git blame argument. I will remove the renaming `asPointer` to `asStringPointer` and add/use the method `asBoolPointer` only where it's necessary. So we'll have `asPointer` and `asBoolPointer` methods in the same file, but less diff changes.